### PR TITLE
fix(pagePlugin): clear some state when delte current page

### PR DESCRIPTION
**Pull request by chilingling**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/plugins/page/src/PageSetting.vue [hunk 1]
```diff
@@ -297,12 +319,20 @@ export default {
...
+                if (pageState?.currentPage?.id !== id) {
+                  return
                 }
-                if (pageState?.currentPage?.id === id) {
-                  emit('openNewPage', pageInfo)
+
+                const staticPageList = (pages || []).find(({ groupId }) => groupId === 0)?.data || []
+                const pageList = collectAllPage(staticPageList)
+
+                const pageHome = pageList.find((page) => page?.isHome)
```
Comment (ghost) [inline packages/plugins/page/src/PageSetting.vue]: 多余的"?"

Comment (chilingling) [reply packages/plugins/page/src/PageSetting.vue]: Done

**Review by ghost — DISMISSED** [review]
LGTM

File: packages/plugins/page/src/PageTree.vue [hunk 2]
```diff
@@ -160,6 +160,17 @@ export default {
     }
 
     const getPageDetail = (pageId) => {
+      if (!`${pageId ?? ''}`) {
```
Comment (wenmine) [inline packages/plugins/page/src/PageTree.vue]: 是否可以直接用 !pageId替代

Comment (chilingling) [reply packages/plugins/page/src/PageTree.vue]: 这个 pageId 是数字，有可能是 0，我可以改成 `pageId !== 0 && !pageId ` 实现更好的可读性

**Review by wenmine — DISMISSED** [review]
LGTM

**Review by gargameljyh — APPROVED** [review]

File: packages/plugins/page/src/PageSetting.vue [hunk 3]
```diff
@@ -288,6 +288,28 @@ export default {
       pageGeneralRef.value.validGeneralForm().then(createHistoryMessage)
     }
 
+    const collectAllPage = (staticPageList = []) => {
+      const pageList = []
+
+      if (!Array.isArray(staticPageList)) {
+        return pageList
```
Comment (ghost) [inline packages/plugins/page/src/PageSetting.vue]: 这个判断分支可以放在定义之前，直接return []

File: packages/plugins/page/src/PageSetting.vue [hunk 4]
```diff
@@ -297,12 +319,20 @@ export default {
...
               pageSettingState.updateTreeData().then((pages) => {
-                const pageInfo = pages?.[0]?.data?.[0] || {
-                  componentName: COMPONENT_NAME.Page
+                if (pageState?.currentPage?.id !== id) {
+                  return
                 }
-                if (pageState?.currentPage?.id === id) {
-                  emit('openNewPage', pageInfo)
+
+                const staticPageList = (pages || []).find(({ groupId }) => groupId === 0)?.data || []
```
Comment (ghost) [inline packages/plugins/page/src/PageSetting.vue]: 魔鬼数字0，不清楚是什么含义，建议用常量定义下

File: packages/plugins/page/src/PageTree.vue [hunk 5]
```diff
@@ -160,6 +160,17 @@ export default {
     }
 
     const getPageDetail = (pageId) => {
+      if (pageId !== 0 && !pageId) {
```
Comment (ghost) [inline packages/plugins/page/src/PageTree.vue:165]: 魔鬼数字0，建议定义常量

Comment (chilingling) [reply packages/plugins/page/src/PageTree.vue:165]: 这里是为了防止 pageId 出现为 0 的情况，没有特定的含义，我加个注释说明一下，就不增加常量存储了

File: packages/plugins/page/src/PageSetting.vue (hunk 4 above)
Comment (chilingling) [reply packages/plugins/page/src/PageSetting.vue]: 这里 0 是代表 静态页面， 1 是代表公共页面，我定一个常量存储一下吧

File: packages/plugins/page/src/PageSetting.vue (hunk 3 above)
Comment (chilingling) [reply packages/plugins/page/src/PageSetting.vue]: Done

**Review by wenmine — APPROVED** [review]

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**